### PR TITLE
Gift Entry: Centralize State Management

### PR DESCRIPTION
**Pull request by kenlewis-sfdo**
[W-8017324](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07B0000008UmDaIAK/view)
# Critical Changes

# Changes

# Issues Closed

# Community Ideas Delivered

# New Metadata

# Deleted Metadata

### CODE REVIEW

File: src/lwc/geFormRenderer/geFormRenderer.js [hunk 1]
```diff
@@ -1766,4 +1724,140 @@ export default class GeFormRenderer extends NavigationMixin(LightningElement) {
...
+        );
+    }
+
+    setInitialValueInFormStateForFieldMappings(fieldMappingDevNames) {
+        this.elementsFor(fieldMappingDevNames)
+            .forEach(el => this.setInitialValueInFormStateForElement(el));
+
+    }
+
+    elementsFor(fieldMappingDevNames) {
```
Comment (psadie-sfdo) [inline src/lwc/geFormRenderer/geFormRenderer.js:1825]: This is returning `undefined` and is throwing a component error when I clear a lookup to the donor account/contact.

**Review by psadie-sfdo — CHANGES_REQUESTED** [review]
Hey Ken, an early issue I'm running into below.

File: src/lwc/geFormRenderer/geFormRenderer.js (hunk 1 above)
Comment (kenlewis-sfdo) [reply src/lwc/geFormRenderer/geFormRenderer.js:1825]: Added the return statement into the function here as we discussed and should be working again now.

**Comment by dfuller114** [comment]
@kenlewis-sfdo I think it will be important to be clear regarding the purpose for each of the state management functions in `geFormRenderer`. Can we add that standard comment blocks for those functions that interact in any way with the form state? Doesn't have to be done immediately if the purpose of those functions will continue to significantly change, but I think it would be helpful to have some detailed descriptions moving forward.

File: src/lwc/geFormField/geFormField.js [hunk 2]
```diff
@@ -636,4 +631,86 @@ export default class GeFormField extends LightningElement {
...
+        if (this.isSiblingRecordTypeIdInFormState) {
+            return this.siblingRecordTypeId;
+        } else if (this.isParentObjectRecordTypeIdInFormState) {
+            return this.parentObjectRecordTypeId;
+        } else {
+            return null;
+        }
+    }
+
+    isParentObjectRecordTypeIdInFormState() {
```
Comment (psadie-sfdo) [inline src/lwc/geFormField/geFormField.js:671]: Out of curiosity, any particular reason these two `Is*` aren't getters?

**Review by psadie-sfdo — CHANGES_REQUESTED** [review]
This is looking good to me! Just a minor comment and a request. I'd also like to take another look once the tests are in place.

File: src/lwc/geFormField/geFormField.js [hunk 3]
```diff
@@ -636,4 +631,86 @@ export default class GeFormField extends LightningElement {
...
+    }
+
+    get fieldMappingsForImportedRecord() {
+        return GeFormService
+            .fieldMappingsForImportedRecordFieldName(this.importedRecordFieldName);
+    }
+
+    get parentObjectRecordTypeId() {
+        return this.parentObjectFromFormState &&
+            this.parentObjectFromFormState .recordTypeId;
```
[the remark this answers is not in the document]
Comment (kenlewis-sfdo) [reply src/lwc/geFormField/geFormField.js:704]: good catch thanks!

File: src/lwc/geFormRenderer/geFormRenderer.js [hunk 4]
```diff
@@ -1766,4 +1745,142 @@ export default class GeFormRenderer extends NavigationMixin(LightningElement) {
...
+    }
+
+    fieldMappingWrapperFor(element) {
+        return element.dataImportFieldMappingDevNames &&
+            GeFormService.getFieldMappingWrapper(element.dataImportFieldMappingDevNames[0]);
+    }
+
+    getValueFrom(element) {
+        return element &&
+        element.recordValue !== undefined ? element.recordValue :
```
[the remark this answers is not in the document]
Comment (kenlewis-sfdo) [reply src/lwc/geFormRenderer/geFormRenderer.js:1882]: I changed this up a little bit to put the truthy and falsy values each on their own line, which I think makes it slightly more readable - thanks for the suggestion!

**Review by ptewson-sfdo — APPROVED** [review]
Looks good!

**Review by psadie-sfdo — APPROVED** [review]
Looks good!

**Comment by kenlewis-sfdo** [comment]
Superseded by [5944](https://github.com/SalesforceFoundation/NPSP/pull/5944)

**Pull request closed without merging**